out/notebooks/spl2/installer.ts

```diff
@@ -0,0 +1,514 @@
+            let pct = Math.floor(totalDownloaded * 100 / totalSize);
+            if (pct === nextUpdate) {
+                progressBar.text = `${progressBarText} ${pct}%`;
+                nextUpdate++;
+            }
+        });
+        fileWriter.on('error', (err) => {
+            error = err;
+            fileWriter.close();
+            reject(err);
+        });
+        fileWriter.on('close', () => {
+            if (!error) {
+                progressBar.hide();
+                resolve(totalSize);
+            }
+        });
+        data.pipe(fileWriter);
+    });
+}
+
+async function extractZipWithProgress(
+    zipfilePath:string,
+    extractPath: string,
+    compressedSize: number,
+    progressBar: StatusBarItem,
+    progressBarText: string,
+): Promise<string> {
+    // Create read and unzip streams and listen for individual entry to find bin\java.exe
+    let binJavaPath;
+    let readCompressedSize = 0;
+    let nextUpdate = 1;
+
+    const pipe = util.promisify(pipeline);
+    const readStream = fs.createReadStream(zipfilePath);
+    const unzipStream = unzipper.Extract({ path: extractPath });
+    unzipStream._writable.on('entry', entry => {
+        if (entry.path.endsWith(path.join('bin', 'java.exe'))) {
+            binJavaPath = path.join(extractPath, entry.path);
+        }
+        readCompressedSize += entry.vars.compressedSize;
+        let pct = Math.floor(readCompressedSize * 100 / compressedSize);
+        if (pct === nextUpdate) {
+            progressBar.text = `${progressBarText} ${pct}%`;
+            nextUpdate++;
+        }
+        entry.autodrain();
+    });
+    await pipe(
+        readStream,
+        unzipStream,
+    );
+    return binJavaPath;
+}
+
+async function extractTgzWithProgress(
+        tgzPath:string,
+        extractPath: string,
+        compressedSize: number,
+        progressBar: StatusBarItem,
+        progressBarText: string,
+    ): Promise<string> {
+    // Create read and unzip streams and listen for individual entry to find bin\java.exe
+    let binJavaPath;
+    let readCompressedSize = 0;
+    let nextUpdate = 1;
+
+    const pipe = util.promisify(pipeline);
+    
+    await pipe(
+        fs.createReadStream(tgzPath).on('data', (chunk) => {
+            readCompressedSize += chunk.length;
+            let pct = Math.floor(readCompressedSize * 100 / compressedSize);
+            if (pct === nextUpdate) {
+                progressBar.text = `${progressBarText} ${pct}%`;
+                nextUpdate++;
+            }
+        }),
+        zlib.createGunzip(),
+        tar.extract(extractPath, {
+            map: (header) => {
+                if (header.name.endsWith(path.join('bin', 'java'))) {
+                    binJavaPath = path.join(extractPath, header.name);
+                }
+                return header;
+            }
+        }),
+    );
+    return binJavaPath;
+}
+
+async function promptToDownloadLsp(alsoInstallJava: boolean): Promise<boolean> {
```

Comment:
Does this work without VPN?

Reply:
yes this is on our public facing jfrog splunk.jfrog.io which is not behind any VPN